js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -192,11 +192,14 @@ export interface IndicesHelper {
 }
 
 const getWgslMappedType = (type: number, components: 1|2|3|4): string|[string, string] => {
+  if (components === 3) {
+    throw new Error('vec3 has same alignment as vec4, use vec4 instead');
+  }
+
```

Comment:
Is it only `vec3<f16>` is invalid, or other types like `vec3<f32>` are also invalid as well?

Reply:
@qjia7

Reply:
Like @dakenf said, vec3 has the same alignment with vec4, we should use it very carefully. Usually, we don't use it as storage type. But as value type, vec3 is still useful. In conv2d-mm, I use it like this https://github.com/microsoft/onnxruntime/blob/main/js/web/lib/wasm/jsep/webgpu/ops/3rd-party/conv2d_mm_webgpu.ts#L41. Storage type is f32, But I need to get 3 contiguous data to return a vec3. I am ok to keep the current changes. Let modify it when we really use this IndicesHelper for vec3 in future.